src/global/config.ts

```diff
@@ -0,0 +1,63 @@
+import { DEFAULT_GLOBAL_CONFIG } from './constants';
+
+type ConfigData = {
+  [key: string]: any;
+};
+
+class Configuration {
+  private static instance: Configuration;
+
+  private config!: ConfigData;
+
+  private constructor() {
+    this.initialize();
+  }
+
+  private initialize(): void {
+    this.config = { ...DEFAULT_GLOBAL_CONFIG };
+  }
+
+  public static getInstance(): Configuration {
+    if (!Configuration.instance) {
+      Configuration.instance = new Configuration();
+    }
+    return Configuration.instance;
+  }
+
+  public get<K extends keyof ConfigData>(
+    key: K,
+    defaultValue?: ConfigData[K]
```

Comment:
The `defaultValue` parameter looks like it doesn't provide much utility.

Reply:
It can be useful sometimes; I'll leave that one.